common/main/java/com/couchbase/lite/internal/BaseImmutableReplicatorConfiguration.java

```diff
@@ -181,6 +182,22 @@ public void addConnectionOptions(@NonNull Map<String, Object> options) {
         final Map<String, Object> httpHeaders = new HashMap<>();
         // User-Agent:
         httpHeaders.put("User-Agent", CBLVersion.getUserAgent());
+
+        // Cookies:
+        final String cookieName = "Cookies";
+        if ((headers != null) && (!headers.isEmpty()) && headers.containsKey(cookieName)) {
```

Comment:
I check for cookies separately to avoid deeply nested if statements

Reply:
Please check for `headers` being null, only once.
... and there is no reason to check to see if it is empty.